### PR TITLE
feat(2022/01)

### PR DESCRIPTION
**Pull request by neverendingqs**

### CODE REVIEW

File: 2022/01/index.js [hunk 1]
```diff
@@ -2,13 +2,44 @@ const { promises: fs } = require('fs');
...
+    .map(inventory => inventory.split('\n').map(calories => parseInt(calories)));
+
+  return inventories;
+}
+
+function sumArray(arr) {
+  return arr.reduce(
+    (acc, element) => acc + element,
+    0
+  );
```
Comment (turtleman270) [inline 2022/01/index.js]: Does the 0 mean your accumulator starts at 0?
Also, why is it split onto 3 lines?

Comment (neverendingqs) [reply 2022/01/index.js]: Yes. One line per argument.

File: 2022/01/index.js [hunk 2]
```diff
@@ -2,13 +2,44 @@ const { promises: fs } = require('fs');
...
+}
+
+async function getSumCalories() {
+  const inventories = await getInput();
+  const sumCalories = inventories.reduce(
```
Comment (faizanminhas) [inline 2022/01/index.js]: This is pretty inefficient, since you’re expanding the array each iteration. This should be .map.

Comment (neverendingqs) [reply 2022/01/index.js]: Oh yeah should've noticed since the initial value is an array :face-palm:

**Comment by neverendingqs** [comment]
Switched to TypeScript.

**Pull request merged**